editor/group_settings_editor.cpp

```diff
@@ -0,0 +1,556 @@
+		_get_all_scenes(p_dir->get_subdir(i), r_list);
+	}
+}
+
+void GroupSettingsEditor::remove_references(const String &p_name) {
+	RBSet<String> scenes;
+	_get_all_scenes(EditorFileSystem::get_singleton()->get_filesystem(), scenes);
+
+	int steps = scenes.size();
+	Vector<EditorData::EditedScene> edited_scenes = EditorNode::get_editor_data().get_edited_scenes();
+	for (const EditorData::EditedScene &es : edited_scenes) {
+		if (!es.root) {
+			continue;
+		}
+		if (es.path.is_empty()) {
+			++steps;
+		}
+	}
+
+	EditorProgress remove("remove_references", TTR("Removing Group References"), steps);
+	int step = 0;
+	// Update opened scenes.
+	Vector<String> edited_scenes_path;
+	for (const EditorData::EditedScene &es : edited_scenes) {
+		if (!es.root) {
+			continue;
+		}
+		remove.step(es.path, step++);
+		bool edited = remove_node_references(es.root, p_name);
+		if (!es.path.is_empty()) {
+			scenes.erase(es.path);
+			if (edited) {
+				edited_scenes_path.push_back(es.path);
+			}
+		}
+	}
+	if (!edited_scenes_path.is_empty()) {
+		EditorNode::get_singleton()->save_scene_list(edited_scenes_path);
+		SceneTreeDock::get_singleton()->get_tree_editor()->update_tree();
+	}
+
+	for (RBSet<String>::Element *E = scenes.front(); E; E = E->next()) {
+		Ref<PackedScene> packed_scene = ResourceLoader::load(E->get());
+		remove.step(E->get(), step++);
+		if (packed_scene->get_state()->remove_group_references(StringName(p_name))) {
+			ResourceSaver::save(packed_scene, E->get());
+		}
+	}
+}
+
+void GroupSettingsEditor::rename_references(const String &p_old_name, const String &p_new_name) {
+	RBSet<String> scenes;
+	_get_all_scenes(EditorFileSystem::get_singleton()->get_filesystem(), scenes);
+
+	int steps = scenes.size();
+	Vector<EditorData::EditedScene> edited_scenes = EditorNode::get_editor_data().get_edited_scenes();
+	for (const EditorData::EditedScene &es : edited_scenes) {
+		if (!es.root) {
+			continue;
+		}
+		if (es.path.is_empty()) {
+			++steps;
+		}
+	}
+
+	EditorProgress rename("rename_references", TTR("Renaming Group References"), steps);
+	int step = 0;
+	// Update opened scenes.
+	Vector<String> edited_scenes_path;
+	for (const EditorData::EditedScene &es : edited_scenes) {
+		if (!es.root) {
+			continue;
+		}
+		rename.step(es.path, step++);
+		bool edited = rename_node_references(es.root, p_old_name, p_new_name);
+		if (!es.path.is_empty()) {
+			scenes.erase(es.path);
+			if (edited) {
+				edited_scenes_path.push_back(es.path);
+			}
+		}
+	}
+	if (!edited_scenes_path.is_empty()) {
+		EditorNode::get_singleton()->save_scene_list(edited_scenes_path);
+		SceneTreeDock::get_singleton()->get_tree_editor()->update_tree();
+	}
+
+	for (RBSet<String>::Element *E = scenes.front(); E; E = E->next()) {
+		Ref<PackedScene> packed_scene = ResourceLoader::load(E->get());
+		rename.step(E->get(), step++);
+		if (packed_scene->get_state()->rename_group_references(StringName(p_old_name), StringName(p_new_name))) {
```

Comment:
```suggestion
		rename.step(E->get(), step++);
		ERR_CONTINUE(packed_scene.is_null());
		if (packed_scene->get_state()->rename_group_references(StringName(p_old_name), StringName(p_new_name))) {
```
The crash happens when scene fails to load for some reason. This should fix it.